prefect_aws/s3.py

```diff
@@ -545,7 +545,10 @@ def _join_bucket_folder(self, bucket_path: str = "") -> str:
                 f"Bucket path {bucket_path!r} is already prefixed with "
                 f"bucket folder {self.bucket_folder!r}; is this intentional?"
             )
-        return (Path(self.bucket_folder) / bucket_path).as_posix()
+
+        return (Path(self.bucket_folder) / bucket_path).as_posix() + (
+            "" if not os.path.join(bucket_path).endswith(os.sep) else os.sep
```

Comment:
Does `os.sep` make sense here? I think we want to explicitly use `/` and not `\` on Windows.

Reply:
Should we consider objects with windows-like path names? 

For eg, if you had objects named like `folderobject\foo.txt`, `folder\object\foo.txt`etc, `os.sep` would make sense no? I don't have a windows machine here but I'm assuming `pathlib` resolves trailing windows-slash chars in the same way it does for unix.

Reply:
Since this is an S3 bucket path, we want to ensure we use `/` because S3 uses POSIX-compliant paths for buckets.
```suggestion
            "" if not os.path.join(bucket_path).endswith("/") else "/"
```

Reply:
Thanks, will update 👍🏽